connectors/service_cli.py

```diff
@@ -40,12 +40,16 @@ async def _start_service(actions, config, loop):
     - performs a preflight check using `PreflightCheck`
     - instantiates a `MultiService` instance and runs its `run` async function
     """
-    preflight = PreflightCheck(config)
+    preflight = PreflightCheck(config, __version__)
     for sig in (signal.SIGINT, signal.SIGTERM):
         loop.add_signal_handler(sig, functools.partial(preflight.shutdown, sig))
     try:
-        if not await preflight.run():
+        success, is_serverless = await preflight.run()
+        if not success:
             return -1
+
+        # injecting this value into the config allows us to avoid checking the server again before requests
+        config["elasticsearch"]["serverless"] = is_serverless
```

Comment:
by putting this in the service cli, will this break if you use Dmitrii's CLI?

Reply:
I tested all the CLI commands and nothing breaks from this.

Reply:
Tested:

- `bin/connectors connector`
- `bin/connectors job`
- `bin/connectors index`

Reply:
you tested those commands against a serverless project? I'm surprised it worked. I must not fully understand how the CLIs interact with one another. I thought those commands would skip this code. 👀

Reply:
> you tested those commands against a serverless project?

Haha.. ha.... one second.

Reply:
I confirmed this works with serverless. I was able to login, create a connector, start a sync job, and delete the index docs without issue.

Reply:
> I thought those commands would skip this code

This code is only run on startup (so `bin/elastic-ingest`). The CLI doesn't really care if something is serverless, it just wants a host and some credentials.

Reply:
ok, looking closer, I think I get it. I was "right", that `bin/connectors` won't set `config["elasticsearch"]["serverless"]`. But since it _also_ doesn't do the preflight check, and that config isn't used anywhere else, it doesn't matter.

Eventually we may need to make sure that we're always aware of if our destination is a Serverless project, but we can cross that bridge when we come to it.

🙈 Sorry for the noise